packages/grid/x-grid/src/tests/editRows.XGrid.test.tsx

```diff
@@ -230,4 +234,105 @@ describe('<XGrid /> - Edit Rows', () => {
+        console.log('handleButtonClick');
+        if (!selectedCellParams) {
+          return;
+        }
+        const { id, field, cellMode } = selectedCellParams;
+        if (cellMode === 'edit') {
+          const editedCellProps = apiRef.current.getEditCellPropsParams(id, field);
+          apiRef.current.commitCellChange(editedCellProps);
+          apiRef.current.setCellMode(id, field, 'view');
+          setButtonLabel('Edit');
+        } else {
+          console.log('Changing cell mode to ', id, field);
+          apiRef.current.setCellMode(id, field, 'edit');
+          setButtonLabel('Save');
+        }
+        // Or you can use the editRowModel prop, but I find it easier
+      }, [selectedCellParams]);
+
+      const handleCellClick = React.useCallback((params: GridCellParams) => {
+        console.log('handleCellClick');
+        setSelectedCellParams(params);
+
+        const { cellMode } = params;
+        if (cellMode === 'edit') {
+          setButtonLabel('Save');
+        } else {
+          console.log('setButtonLabel Edit', params.value);
+          setButtonLabel('Edit');
+        }
+      }, []);
+
+      const handleDoubleCellClick = React.useCallback(
+        (params: GridCellParams, event: React.SyntheticEvent) => {
+          event.stopPropagation();
+        },
+        [],
+      );
+
+      // Prevent from rolling back on escape
+      const handleCellKeyDown = React.useCallback((params, event: React.KeyboardEvent) => {
+        if (
+          params.cellMode === 'edit' &&
+          (event.key === 'Escape' || event.key === 'Delete' || event.key === 'Enter')
+        ) {
+          event.stopPropagation();
+        }
+      }, []);
+
+      // Prevent from committing on blur
+      const handleCellBlur = React.useCallback((params, event?: React.SyntheticEvent) => {
+        if (params.cellMode === 'edit') {
+          event?.stopPropagation();
+        }
+      }, []);
+
+      return (
+        <div style={{ height: 450, width: '100%' }}>
+          <Button className="edit-button" onClick={handleButtonClick} color="primary">
+            {buttonLabel}
+          </Button>
+          <div style={{ height: 400, width: '100%' }}>
+            <XGrid
+              {...baselineProps}
+              apiRef={apiRef}
+              onCellClick={handleCellClick}
+              onCellDoubleClick={handleDoubleCellClick}
+              onCellBlur={handleCellBlur}
+              onCellKeyDown={handleCellKeyDown}
+            />
+          </div>
+        </div>
+      );
+    }
+
+    render(<TestComponent />);
+    fireEvent.click(getCell(0, 0));
+
+    fireEvent.click(screen.getByRole('button', { name: /Edit/i }));
+    fireEvent.click(getCell(1, 0));
+    fireEvent.click(screen.getByRole('button', { name: /Edit/i }));
+    const inputs = document.querySelectorAll('input');
+    expect(inputs.length).to.equal(2);
+    const nikeInput = getCell(0, 0).querySelector('input');
+    fireEvent.click(nikeInput);
```

Comment:
@oliviertassinari any idea why this test does not work?

Reply:
As for why the test case doesn't work. I recall that your first commit was to add a blur listener to fix the issue, so if you don't handle the activeElement, I don't see how the test could work, e.g. calling .focus(). We are inside a browser, fire event click doesn't trigger all the intermediary event (mouse down/mouse up/focus) that a real browse have. You have to simulate them.